transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/TransactionOutboxListener.java

```diff
@@ -52,13 +52,14 @@ default void failure(TransactionOutboxEntry entry, Throwable cause) {
 
   /**
    * Fired when a transaction outbox task has passed the maximum number of retries and has been
-   * blacklisted. This event is not guaranteed to fire in the event of a JVM failure or power loss.
-   * It is fired <em>after</em> the commit to the database marking the task as blacklisted.
+   * designated as a failed task. This event is not guaranteed to fire in the event of a JVM failure
+   * or power loss. It is fired <em>after</em> the commit to the database marking the task as
+   * failed.
    *
-   * @param entry The outbox entry blacklisted.
+   * @param entry The outbox entry to be marked as failed.
    * @param cause The cause of the most recent failure.
    */
-  default void blacklisted(TransactionOutboxEntry entry, Throwable cause) {
+  default void markFailed(TransactionOutboxEntry entry, Throwable cause) {
```

Comment:
We're mixing terminology in this listener - we have `failure` and `markFailed`. We should standardise on what the term "failed" means. What's it called when an attempt fails?

How about `attemptFailed` and `permanentlyFailed`?